frontend/src/utils/api/api.ts

```diff
@@ -0,0 +1,45 @@
+/* eslint-disable */
+
+import axios from 'axios';
+import { APIEndpoints } from './types/endpoints';
+import composeEndpoint from './composeEndpoint';
+
+const API: APIEndpoints = {
+  planner: {
+    validate: (plannerData) => axios.post('/planner/validateTermPlanner', plannerData),
+  },
+
+  courses: {
+    jsonified: (courseCode) => axios.get(`/courses/jsonified/${courseCode}`),
```

Comment:
can we rename the keys for the different endpoints?

Like for example, if the endpoint was `/courses/getCourse/`, the object should be `courses: { getCourse: ..., ... }`

So that its more obvious which api call we are calling i.e. `API.courses.getCourse` over `API.courses.course`